src/Orchard/Indexing/IIndexDocument.cs

```diff
@@ -11,7 +11,7 @@ public interface IDocumentIndex {
         IDocumentIndex Add(string name, int value);
         IDocumentIndex Add(string name, bool value);
         IDocumentIndex Add(string name, double value);
-
+        IDocumentIndex Add(string name, object value);
```

Comment:
I kinda feel, the below implementation should be removed from LuceneDocumentIndex.cs. I prefer letting the caller decide how the data type should be handled within the index document. Anyway all it does is stringyfying the object.
```
public IDocumentIndex Add(string name, object value) {
            return Add(name, value.ToString());
        }
```